braindecode/models/eegconformer.py

```diff
@@ -0,0 +1,272 @@
+# Authors: Yonghao Song <eeyhsong@gmail.com>
+#
+# License: BSD (3-clause)
+
+import torch
+import torch.nn.functional as F
+from einops import rearrange
+from einops.layers.torch import Rearrange, Reduce
+from torch import nn, Tensor
+
+
+class EEGConformer(nn.Sequential):
+    """EEG Conformer.
+
+    Convolutional Transformer for EEG decoding.
+
+    The paper and original code with more details about the methodological
+    choices are availible at the [EEG Conformer]_ and [EEG Conformer Code]_.
+
+    This neural network architecture recieves a different input shape than
+    other models. The input shape should be four-dimensional matrix
+    representing the EEG signals.
+
+         (batch_size, n_channels, n_timesteps)`.
+
+    The EEG Conformer architecture is composed of three modules:
+        - PatchEmbedding
+        - TransformerEncoder
+        - ClassificationHead
+
+    Notes
+    -----
+    The authors recommend using augment data before using Conformer, e.g. S&R,
+    at the end of the code.
+    Please refer to the original paper and code for more details.
+
+    .. versionadded:: 0.8
+
+    We aggregate the parameters based on the parts of the models.
+
+    Parameters PatchEmbedding
+    -------------------------
+    - n_kernel: int
+        Number of kernels for the temporal convolution layer (first layer).
+    - n_kernel_temp_conv: int
```

Comment:
if we speak about kernel size I would prefer notation with `len` in the name. `n_kernel_temp_conv` and other like that may suggest that this corresponds to the number of kernels

Reply:
Similar thing for the `spatial_conv` and pooling layers. I would recommend to keep similar naming to the parameters in `ShallowFBCSPNet` as it is almost the same (also I've found strong link to the `ShallowFBCSPNet` in the article)

Reply:
I think I solved this name variable issue. Can you please check @sliwy?